.github/workflows/build-and-test.yml

```diff
@@ -296,7 +296,7 @@ jobs:
 
     # Set custom deployment target because of rocksdb
     - name: Set deployment target (macOS)
-      run: echo "MACOSX_DEPLOYMENT_TARGET=10.14" >> $GITHUB_ENV 
+      run: echo "MACOSX_DEPLOYMENT_TARGET=10.12" >> $GITHUB_ENV 
```

Comment:
I recall getting errors with 10.12, but I'll double check to see why this isn't happening anymore:

```
db/version_set.cc:5900:53: error: 'value' is unavailable: introduced in macOS 10.13
              cfd->user_comparator()->Compare(start.value(),
                                                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/optional:950:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
db/version_set.cc:5908:51: error: 'value' is unavailable: introduced in macOS 10.13
              cfd->user_comparator()->Compare(end.value(),
                                                  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/optional:950:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
```

This was with RocksDB though, not `rust-rocksdb`, so maybe that's the issue

Reply:
@rajivshah3 any news on this?

Reply:
Seems like this is still the case for the Node.js bindings. I'm not sure why it's not happening with the Python bindings on CI. There are two groups of errors when I compile the Node.js bindings locally:

1. `introduced in macOS 10.13`: this seems right. The good thing is that based on [this website](https://www.macworld.com/article/673697/what-version-of-macos-can-my-mac-run.html), every computer that supported 10.12 (Sierra) supports 10.13 (High Sierra) so users should be able to update their OS
2. I'm getting this error again:
```
warning: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__memory/unique_ptr.h:88:5: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
```
This is the issue with aligned allocation/deallocation in https://github.com/iotaledger/wallet.rs/issues/1635 (it's actually supported in macOS 10.13 and Clang is incorrectly giving an error). I emailed Apple Developer Support about this in December and I don't believe I received a response. I may try posting on their forums because some of the employees are usually pretty responsive. Setting `-faligned-new` seems to silence the error but I'm not sure what happens if we do that

Reply:
> I'm not sure why it's not happening with the Python bindings on CI

I'm seeing similar errors when building the Python bindings locally, so I think something is cached on CI with the newer macOS target

Reply:
As far as I understood this flag changes the way dynamical objects aligned in the memory. Enabling it instructs the compiler to automatically align dynamically allocated objects to the maximum alignment for the target platform.  I think the worst thing that could happen, is the app becoming slightly larger. What is your opinion on trying out the flag and see whether it solves the issues people seem to be having?

Reply:
Should we go ahead and merge or are we still investigating this?

Reply:
Python bindings seems to fail to compile